libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -804,6 +803,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         restartRouteScope()
         threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
             routeUpdateMutex.withLock {
+                navigationSession.setRoutes(routes)
```

Comment:
This is not about 1TAP, but rather Copilot, which is a separate library. This change allows Copilot to automatically start recording Active Guidance sessions. Otherwise we will have to provide an extra function for that and make each client call it before starting a trip. If we try to automatically start recording Active Guidance sessions without this change, history files won't contain SET_ROUTE event. We could retrieve this event from the context part of the file in this case, but nav-native has to add support for that first.

Reply:
Agreed let's cut a separate ticket as both OP and #5934 are closed 🙏